docs/modules/architecture/pages/compact-binary-specification.adoc

```diff
@@ -0,0 +1,706 @@
+|optional<std::vector<optional<hazelcast:client:offset_date_time>>> 
+|HOffsetDateTime?[]?
+|Optional[list[Optional[datetime.datetime]]]
+|(OffsetDateTime \| null)[] \| null
+|[]*types.OffsetDateTime
+|
+|Array of TimestampWithTimeZones
+|no
+|compact[]
+|T[] 
+|template<typename T> optional<std::vector<optional<T>>> 
+|T?[]?
+|Optional[list[Optional[Any]]]
+|(T \| null)[] \| null
+|[]interface{}
+|
+|Array of compacts
+|no
+|nullable-boolean
+|Boolean
+|optional<bool>
+|bool?
+|Optional[bool]
+|boolean \| null
+|*bool
+|
+|null or 
+int8 1 for true
+int8 0 for false
+|no
+|nullable-int8
+|Byte
+|optional<int8_t>
+|sbyte?
+|Optional[int]
+|number \| null
+|*int8
+|
+|An int8 that can also be null
+|no
+|nullable-int16
+|Short
+|optional<int16_t>
+|short?
+|Optional[int]
+|number \| null
+|*int16
+|
+|An int16 that can also be null
+|no
+|nullable-int32
+|Integer
+|optional<int32_t>
+|int?
+|Optional[int]
+|number \| null
+|*int32
+|
+|An int32 that can also be null
+|no
+|nullable-int64
+|Long
+|optional<int64_t>
+|long?
+|Optional[int]
+|Long \| null
+|*int64
+|
+|An int64 that can also be null
+|no
+|nullable-float32
+|Float
+|optional<float>
+|float?
+|Optional[float]
+|number \| null
+|*float32
+|
+|A float32 that can also be null
+|no
+|nullable-float64
+|Double
+|optional<double>
+|double?
+|Optional[float]
+|number \| null
+|*float64
+|
+|A double that can also be null
+|no
+|nullable-boolean[]
+|Boolean[] 
+|optional<std::vector<optional<bool>>> 
+|bool?[]?
+|Optional[list[Optional[bool]]]
+|(boolean \| null)[] \| null
+|[]*bool
+|
+|Array of nullable booleans
+|no
+|nullable-int8[]
+|Byte[] 
+|optional<std::vector<optional<int8_t>>> 
+|sbyte?[]?
+|Optional[list[Optional[int]]]
+|(number \| null)[] \| null
+|[]*int8
+|
+|Array of nullable int8s
+|no
+|nullable-int16[]
+|Short[] 
+|optional<std::vector<optional<int16_t>>>
+|short?[]?
+|Optional[list[Optional[int]]]
+|(number \| null)[] \| null
+|[]*int16
+|
+|Array of nullable i1int6s
+|no
+|nullable-int32[]
+|Integer[] 
+|optional<std::vector<optional<int32_t>>>
+|int?[]?
+|Optional[list[Optional[int]]]
+|(number \| null)[] \| null
+|[]*int32
+|
+|Array of nullable int32s
+|no
+|nullable-int64[]
+|Long[] 
+|optional<std::vector<optional<int64_t>>> 
+|long?[]?
+|Optional[list[Optional[int]]]
+|(Long \| null)[] \| null
+|[]*int64
+|
+|Array of nullable int64s
+|no
+|nullable-float32[]
+|Float[] 
+|optional<std::vector<optional<float>>> 
+|float?[]?
+|Optional[list[Optional[float]]]
+|(number \| null)[] \| null
+|[]*float32
+|
+|Array of nullable float32s
+|no
+|nullable-float64[]
+|Double[] 
+|optional<std::vector<optional<double>>> 
+|double?[]?
+|Optional[list[Optional[float]]]
+|(number \| null)[] \| null
+|[]*float64
+|
+|Array of nullable float64
+|no
+|===
+
+
+=== Type Ids
+
+Each type supported in the wire format has its type id. The type ids are used while constructing the schemas, performing type checks for user access to fields, and are exposed to the user as a public API.
```

Comment:
=== Type IDs

Each type supported in the wire format has its type ID. The type IDs are used while constructing the schemas, performing type checks when accessing to fields, and are exposed as a public API.